### ISSUE TITLE
Import shared styl files to import individual components.

### ISSUE THREAD

**Issue by M-O-Z-G** [groenroos/bulma-stylus#24]
I found that Bulma.io team starts to import some common files into separate component and elements files. For example to [`button`](https://github.com/jgthms/bulma/blob/master/sass/elements/button.sass). This allows these files to be imported individually into a user's project.
The problem is that if we use it with Stylus, we meets duplication of some virtual rules like `$controls` spread around the resulting CSS file. I see, you didn't make these changes at all in the latest release. Perhaps due the problem.

[I made a lifehack](https://github.com/M-O-Z-G/bulma-stylus-plus/tree/bsp-091b) to prevent duplication and still allow to use files separately, with variables (name is generated randomly, to prevent any conflicts).

**Example:**

**control.styl**:
```styl
00: $_goSyMvit := true;
...
```

**button.styl**:
```styl
00: unless $_goSyMvit is defined and $_goSyMvit
01:	@import "../utilities/controls"
02:
03: @import "../utilities/mixins"
...
```

**Comment by groenroos** [comment]
Thanks for doing this legwork! 👍 

Since I just want to mirror the current release of Bulma, I worked off of [their 0.9.1 tag](https://github.com/jgthms/bulma/tree/0.9.1) where they don't yet have this functionality. I don't want to start translating the changes in their master until they release it, since there'd be little visible benefit to it, and there's a high risk of duplicate work as it's a very active project with fixes and changes committed frequently.

However, once they do release 0.9.2, I suspect this'll become in handy. So thanks for the snippet, I'll keep it in mind, thank you!

This may be too much to ask, but do you reckon there's a way to do this that would still be a single line? Maybe taking the logic into its own import or something? Reason I ask is I really want to keep the line numbers between Stylus and Sass the same as much as possible, since it makes it easier to compare and spot omissions.

**Comment by M-O-Z-G** [comment]
> do this that would still be a single line

**My mistake**, sorry for it:( We should use [`@require`](https://stylus-lang.com/docs/import.html#require) instead [`@import`](https://stylus-lang.com/docs/import.html#stylus-import).

[Example in my 0.9.1 branch.](https://github.com/M-O-Z-G/bulma-stylus-plus/commit/3535ffee94a906c1192e010190abc379f089e502)

**Comment by groenroos** [comment]
Good point - though I haven't looked into it too deeply, it looks like the original Bulma uses `@import`, which I guess would have [the same problem](https://sass-lang.com/documentation/at-rules/import)? In which case, I'd have to consider it a "feature" of the original Bulma.

Out of interest, is there any particular reason you've converted only some of the `@import`s into `@require`s in your repo? Feels like the benefit of one-time importing would be good to have consistently across the board, no?

**Comment by M-O-Z-G** [comment]
> In which case, I'd have to consider it a "feature" of the original Bulma.

SASS can handle its virtual rules differently from Stylus, which is most likely it is, because duplicate code with different set of selectors doesn't appear in the original. But I have not done specific tests with SASS, so will not say for sure how it works there.

> Out of interest, is there any particular reason you've converted only some of the `@import`s into `@require`s in your repo?

Just didn't fix what didn't break.
